lib/ansible/modules/cloud/azure/azure_rm_virtualmachineextension_facts.py

```diff
@@ -0,0 +1,248 @@
+            sample: myvm
+        name:
+            description:
+                - Virtual machine name
+            returned: always
+            type: str
+            sample: myextension
+        location:
+            description:
+                - Location
+            returned: always
+            type: str
+            sample: eastus
+        publisher:
+            description:
+                - Extension publisher
+            returned: always
+            type: str
+            sample: Microsoft.Azure.Extensions
+        type:
+            description:
+                - Extension type
+            returned: always
+            type: str
+            sample: CustomScript
+        settings:
+            description:
+                - Extension specific settings dictionary
+            returned: always
+            type: complex
+            sample: "{'commandToExecute': 'hostname'}"
+        auto_upgrade_minor_version:
+            description:
+                - Autoupgrade minor version flag
+            returned: always
+            type: bool
+            sample: true
+        tags:
+            description:
+                - Resource tags
+            returned: always
+            type: complex
+            sample: "{ mytag: abc }"
+        provisioning_state:
+            description:
+                - Provisioning state of the extension
+            returned: always
+            type: str
+            sample: Succeeded
+'''
+
+from ansible.module_utils.azure_rm_common import AzureRMModuleBase
+
+try:
+    from msrestazure.azure_exceptions import CloudError
+    from msrest.serialization import Model
+    from azure.mgmt.compute import ComputeManagementClient
+except ImportError:
+    # This is handled in azure_rm_common
+    pass
+
+
+class AzureRMVirtualMachineExtensionFacts(AzureRMModuleBase):
+    def __init__(self):
+        # define user inputs into argument
+        self.module_arg_spec = dict(
+            resource_group=dict(
+                type='str',
+                required=True
+            ),
+            virtual_machine_name=dict(
+                type='str',
+                required=True
+            ),
+            name=dict(
+                type='str'
+            ),
+            tags=dict(
+                type='list'
+            )
+        )
+        # store the results of the module operation
+        self.results = dict(
+            changed=False
+        )
+        self.resource_group = None
+        self.virtual_machine_name = None
+        self.name = None
+        self.tags = None
+        self.mgmt_client = None
+        super(AzureRMVirtualMachineExtensionFacts, self).__init__(self.module_arg_spec, supports_tags=False)
+
+    def exec_module(self, **kwargs):
+        for key in self.module_arg_spec:
+            setattr(self, key, kwargs[key])
+
+        # need newer mgmt client that supports list
+        self.mgmt_client = self.get_mgmt_svc_client(ComputeManagementClient,
```

Comment:
there's compute_client already in common, pls use it for consistent, otherwise, some modules use common client, some use private. it will cause in-consistent issue when  version upgrading. for new modules has no mgmt client in common,  it's ok to add it in module.